File: CodingSamples/Java/Runtime/ReflectionTest1/Program1.java
class Program {

	private static void present(Item info) {
		System.out.println("<Item>");
		System.out.printf("  <name>%s</name>%n", info.name());
		System.out.printf("  <brand>%s</brand>%n", info.brand());
		System.out.println("</Item>");
	}

	private static void present(Customer info) {
		System.out.println("<Customer>");
		System.out.printf("  <id>%s</id>%n", info.id());
		System.out.printf("  <purchase>%s</purchase>%n", info.purchase());
		System.out.printf("  <rating>%s</rating>%n", info.rating());
		System.out.println("</Customer>");
	}

	public static void main(String[] args) {
		present(Shop.getCommonItem());
		System.out.println();
		present(Shop.getBestCustomer());
	}
}

File: CodingSamples/DotNet/Runtime/ReflectionTest1/DemoApp/Program1.cs
class Program
{
    static void Present(Item info)
    {
        Console.WriteLine("<Item>");
        Console.WriteLine("  <Name>{0}</Name>", info.Name);
        Console.WriteLine("  <Brand>{0}</Brand>", info.Brand);
        Console.WriteLine("</Item>");
    }

    static void Present(Customer info)
    {
        Console.WriteLine("<Customer>");
        Console.WriteLine("  <Id>{0}</Id>", info.Id);
        Console.WriteLine("  <Purchase>{0}</Purchase>", info.Purchase);
        Console.WriteLine("  <Rating>{0}</Rating>", info.Rating);
        Console.WriteLine("</Customer>");
    }
    
    static void Main(string[] args)
    {
        Present(Shop.BestItem);
        Console.WriteLine("---------------");
        Present(Shop.BestCustomer);
    }
}
